### PR TITLE
[CALCITE-5882] Compile-time evaluation of SPLIT function returns incorrect result

### PR DESCRIPTION
**Pull request by mihaibudiu**

### CODE REVIEW

**Comment by mihaibudiu** [comment]
@tanclary I think you have introduced the SPLIT function.

**Comment by rubenada** [comment]
Thanks for this patch @mihaibudiu . I think your analysis in the Jira ticket and your proposed fix are correct.
In any case, since @tanclary was the one that implemented the SPLIT function, I'll let some time to see if he wants to chime in and make any comments in the PR.

**Review by NobiGo — APPROVED** [review]
LGTM

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java [hunk 1]
```diff
@@ -5090,6 +5091,19 @@ private HepProgram getTransitiveProgram() {
         .checkUnchanged();
   }
 
+  /** Test case for <a href="https://issues.apache.org/jira/browse/CALCITE-5882">
+   * [CALCITE-5882] Compile-time evaluation of SPLIT function returns incorrect result</a>. */
+  @Test public void testSplit() {
```
Comment (tanclary) [inline core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java:5096]: Is there a reason this test is in RelOptRules? My understanding is this test class is for optimization rules, I am not sure if a function bug test necessarily belongs here. Maybe the operator tests would work better? Let me know if I am missing something.

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -332,6 +332,11 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
           Static.RESOURCE.delimiterIsRequired(
               operatorBinding.getOperator().getName(), type.toString()));
     }
+    if (SqlTypeUtil.isBinary(type)) {
+      type = operatorBinding.getTypeFactory().createSqlType(SqlTypeName.VARBINARY);
```
[the remark this answers is not in the document]
Comment (mihaibudiu) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: I have rewritten it this way.

**Comment by rubenada** [comment]
@mihaibudiu could you please resolve the conflicts? (probable caused by my recent merge of CALCITE-5879)

 @tanclary do you have any other remark here? Or shall I move on with the merge once the conflicts are solved?

**Comment by tanclary** [comment]
> @mihaibudiu could you please resolve the conflicts? (probably caused by my recent merge of [CALCITE-5879](https://issues.apache.org/jira/browse/CALCITE-5879))
> 
> @tanclary do you have any other remark here? Or shall I move on with the merge once the conflicts are solved?

@rubenada Looks good to me, thanks for helping with the review!

**Pull request merged**